app/Global.java

```diff
@@ -80,11 +97,46 @@ private static void fixJavaKerberos() {
 
   }
 
+  /**
+   * This method is overridden to re-direct all the requests to HTTP endpoint of Dr.Elephant to
+   * the respective HTTPS enabled endpoint(If such endpoints exists)
+   * @param requestHeader Request Header
+   * @return The Handler to appropriately handle the given request
+   */
+  @Override
+  public Handler onRouteRequest(Http.RequestHeader requestHeader) {
+    if (!Strings.isNullOrEmpty(HTTP_PORT) && !Strings.isNullOrEmpty(HTTPS_PORT)) {
+      String port = INVALID_PORT;
+      if (!Strings.isNullOrEmpty(requestHeader.host())) {
+        port = getPortFromHostAddress(requestHeader.host());
+      }
+      if (!port.equals(INVALID_PORT) && port.equals(HTTP_PORT)) {
+        return controllers.Default.redirect(HTTPS_PROTOCOL + requestHeader.host().replace(port, HTTPS_PORT)
+            + requestHeader.uri());
+      }
+    }
+    return super.onRouteRequest(requestHeader);
+  }
+
   static void setFinalStatic(Field field, Object newValue) throws Exception {
     field.setAccessible(true);
     Field modifiersField = Field.class.getDeclaredField("modifiers");
     modifiersField.setAccessible(true);
     modifiersField.setInt(field, field.getModifiers() & ~Modifier.FINAL);
     field.set(null, newValue);
   }
+
+  /**
+   *
+   * @param hostAddress The hostAddress string which will be of format "elephant.abc.com:8080"
+   * @return Return the port of the of the hostAddress if exists else return INVALID_PORT
+   */
+  private String getPortFromHostAddress(String hostAddress) {
+    String patterForPort = "(.+):([\\d]{1,5})";
+    Matcher matcher = Pattern.compile(patterForPort).matcher(hostAddress);
+    if (matcher.find()) {
+      return matcher.group(2);
+    }
+    return INVALID_PORT;
+  }
```

Comment:
You might want to consider leveraging [Guava's `HostAndPort.fromString()`](https://guava.dev/releases/18.0/api/docs/com/google/common/net/HostAndPort.html#fromString(java.lang.String)) instead of manually implementing this logic. I also don't think this regular expression will work if it's a bare IPv6 address.

Reply:
Big thanks Logan, I was looking for such a utility method and couldn't find it so implemented a crude one. Thanks for the review and for reporting this issue.